### PR TITLE
admin can destroy events and organization can cancel events

### PR DESCRIPTION
**Pull request by Omarkojak**
for issue #134

### CODE REVIEW

**Comment by Omarkojak** [comment]
@AhmadElsagheer

File: app/Http/Controllers/Event/EventController.php [hunk 1]
```diff
@@ -109,11 +111,20 @@ public function update(EventRequest $request, $id)
 	}
 
 	/**
-	 * Cancel an event.
+	 * Organization cancel an event
+	 */
+	public function cancel($id)
+	{
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php:125]: You must check here that the authenticated organization is the one who created the event

**Comment by Omarkojak** [comment]
@AhmadElsagheer

File: app/Http/Controllers/Event/EventController.php [hunk 2]
```diff
@@ -36,9 +36,11 @@ public function __construct()
         ]]);
 
         $this->middleware('auth_organization', ['only' => [
-            'create', 'store', 'edit', 'update', 'destroy',
+            'create', 'store', 'edit', 'update', 'cancel'
         ]]);
-    }
+
+		$this->middleware('auth_admin', ['only' => ['destroy']]);
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php]: fix identation

File: app/Http/Controllers/Event/EventController.php [hunk 3]
```diff
@@ -116,12 +118,24 @@ public function update(EventRequest $request, $id)
 	}
 
 	/**
-	 * Cancel an event.
-	 * @param int $id event id.
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php:120]: add this line below the `Organization cancel an event`

File: app/Http/Controllers/Event/EventController.php [hunk 4]
```diff
@@ -116,12 +118,24 @@ public function update(EventRequest $request, $id)
...
+	 * Organization cancel an event
 	 */
+	public function cancel($id)
+	{
+		if(auth()->guard('organization')->user()->id == Event::findorfail($id)->organization()->id)
+		{
+			$this->eventService->cancel($id);
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php]: Pass the organization to the cancel function as this function is used by both web and API. Checking for the authenticated organization must be done in the service. The service must return true or false.

File: app/Http/Services/EventService.php [hunk 5]
```diff
@@ -56,15 +56,24 @@ public function update(EventRequest $request, $id)
...
-		}
+		$event->delete();
+		Notification::notify($event->volunteers, 3, null,
+			"Event ".($event->name)."has been cancelled", url("/"));
+		$this->unindexEvent($event->id);
+	}
+
+	/**
+	 * Delete an event.
+	 */
```
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php:76]: Add `@param` to indicate what this `id` is

File: app/Http/Services/EventService.php [hunk 6]
```diff
@@ -56,15 +56,24 @@ public function update(EventRequest $request, $id)
...
+	public function destroy($id)
+	{
+		$event = Event::findOrFail($id);
+		$event->forceDelete();
+		Notification::notify($event->volunteers, 3, null,
```
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php:81]: You need to send the notification and unindex the event before deletion?

File: app/Http/routes.php [hunk 7]
```diff
@@ -251,6 +251,11 @@
     Route::get('event/{id}/review/{r_id}/report', 'Event\EventReviewController@report');
     Route::resource('event/{id}/review','Event\EventReviewController');
 
+    /*
+     * Event Cancelling
+     */
```
Comment (AhmadElsagheer) [inline app/Http/routes.php]: remove doc and add it below CRUD doc but above the resource

**Comment by AhmadElsagheer** [comment]
Need to handle `cancel` function only in `EventAPIController`

File: app/Http/Controllers/Event/EventController.php (hunk 2 above)
Comment (Omarkojak) [inline app/Http/Controllers/Event/EventController.php]: ana 3ndy mzbota msh 3arf leh :D

File: app/Http/Services/EventService.php (hunk 6 above)
Comment (Omarkojak) [inline app/Http/Services/EventService.php:81]: maybe i don't need to unindex but i need to notify and that was stated in the issue

Comment (Omarkojak) [inline app/Http/Services/EventService.php:81]: i tested it and we needed unindexing the event too

**Comment by Omarkojak** [comment]
@AhmadElsagheer

File: app/Http/Services/EventService.php (hunk 6 above)
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php:81]: I mean do you need it before or after the delete call? Is it working fine?

**Pull request merged**